compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/AnnotationGenerator.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.backend.generators
+
+import org.jetbrains.kotlin.descriptors.annotations.AnnotationUseSiteTarget
+import org.jetbrains.kotlin.fir.FirAnnotationContainer
+import org.jetbrains.kotlin.fir.backend.Fir2IrVisitor
+import org.jetbrains.kotlin.fir.declarations.FirProperty
+import org.jetbrains.kotlin.fir.declarations.FirValueParameter
+import org.jetbrains.kotlin.ir.declarations.*
+import org.jetbrains.kotlin.ir.expressions.IrConstructorCall
+import org.jetbrains.kotlin.ir.util.isPropertyField
+import org.jetbrains.kotlin.ir.util.isSetter
+
+internal class AnnotationGenerator(private val visitor: Fir2IrVisitor) {
+
+    fun generate(irContainer: IrMutableAnnotationContainer, firContainer: FirAnnotationContainer) {
+        irContainer.annotations = firContainer.annotations.mapNotNull {
```

Comment:
Just curious if we would ever expect the `it.accept(...)` call to return `null` or something other than an `IrConstructorCall`. What case would that be? If not, then maybe we should just assert that here.

Reply:
Hm, I guess such other cases would indicate compilation/resolution errors at earlier stages, but not 100% sure. @mglukhikh will have better answers. :)

Reply:
It can return `IrErrorCallExpression` in case of resolution errors.

Reply:
Sounds like the current handling is needed, then.